geoportal/c2cgeoportal_geoportal/views/login.py

```diff
@@ -155,15 +166,47 @@ def login(self):
                             {
                                 "username": user.username,
                                 "is_password_changed": False,
-                                "two_factor_enable": True,
+                                "two_factor_enable": self.two_factor_auth,
                             }
                         ),
                         headers=(("Content-Type", "text/json"),),
                     ),
                 )
 
-            headers = remember(self.request, username)
             LOG.info("User '%s' logged in.", username)
+            if self.request.GET.get("type") == "oauth2":
+                self.request.user_ = user
+                LOG.debug(
+                    "Call OAuth create_authorization_response with:\nurl: %s\nmethod: %s\nbody:\n%s",
+                    self.request.current_route_url(_query=self.request.GET),
+                    self.request.method,
+                    self.request.body,
+                )
+                headers, body, status = oauth2.get_oauth_client(
+                    self.request.registry.settings
+                ).create_authorization_response(
+                    self.request.current_route_url(_query=self.request.GET),
+                    self.request.method,
+                    self.request.body,
+                    self.request.headers,
+                )
+                if hasattr(self.request, "tm"):
+                    self.request.tm.commit()
+                LOG.debug("OAuth create_authorization_response return\nstatus: %s\nbody:\n%s", status, body)
+
+                if status == 302:
+                    raise HTTPFound(location=headers["Location"])
+                if status != 200:
+                    if body:
+                        raise exception_response(status, details=body)
+                    raise exception_response(status)
+                return set_common_headers(
+                    self.request,
+                    "login",
+                    NO_CACHE,
+                    response=Response(body, headers=headers.items()),
+                )
+            headers = remember(self.request, username)
```

Comment:
This is a huge function now. Hope we will be able to cut it at some time.
Won't it make sense to create a dedicated enpoint for OAuth ?

Reply:
I extract the function, I'm not certain if that's a good way to create a dedicated enpoint, I don't want to duplicate code for that...